internal/validation/validate.go

```diff
@@ -255,6 +256,17 @@ func (v *APIRuleValidator) validateAccessStrategies(attributePath string, access
 	for i, r := range accessStrategies {
 		strategyAttrPath := attributePath + fmt.Sprintf("[%d]", i)
 		problems = append(problems, v.validateAccessStrategy(strategyAttrPath, r, selector, namespace)...)
+		switch r.Handler.Name {
```

Comment:
I think the logic we have implemented here is not correct, because it allows combination that should be restricted, e.g. "allow" and "noop" on same rule. 

I remembered that we have a decision about that. Obviously no one remember about that, when I asked it in Slack and when we were discussing this in the refinement. 😁 I checked the decision we have in [api-gateway](https://github.com/kyma-project/api-gateway/issues/329). And this decision is actual different then what we have in the docs. As far as I understand the decision, noop is allowed to be on the same path with OAuth2 and Ory JWT, but I think this doesn't make sense and is incorrect in the ADR.

Also the validation is different depending on if you are using JWT or ORY handler, so we shouldn't mix this with general validation.
There is already a validation implemented for the other validations in [istio as_validator](https://github.com/kyma-project/api-gateway/blob/main/internal/processing/istio/as_validator.go) and [ory as_validator](https://github.com/kyma-project/api-gateway/blob/main/internal/processing/ory/as_validator.go).